src/Plugin/Field/FieldWidget/TranslationSynchronisationWidget.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_translation\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageInterface;
+
+/**
+ * Defines the 'Translation Synchronisation Widget' field widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_translation_translation_sync_widget",
+ *   label = @Translation("Translation Synchronisation Widget"),
+ *   field_types = {"oe_translation_translation_sync"},
+ * )
+ */
+class TranslationSynchronisationWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element += [
+      '#type' => 'fieldset',
+    ];
+
+    $element['type'] = [
+      '#type' => 'select',
+      '#empty_option' => $this->t('None'),
+      '#description' => $this->t('Select the type of synchronization rule.'),
+      '#title' => $this->t('Select type'),
+      '#options' => static::getSyncTypeOptions(),
+      '#default_value' => isset($items[$delta]->type) ? $items[$delta]->type : NULL,
+    ];
+
+    // Get the selector to build the paths for type field.
+    $parents = $element['#field_parents'];
+    $parents[] = $this->fieldDefinition->getName();
+    $parents[] = $element['#delta'];
+    $selector = array_shift($parents);
+    if ($parents) {
```

Comment:
I don't understand this very well: previously we assigned two new elements to the $parents array so even if we shift the first one the second will always be there so $parents will never be empty.